pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,726 @@
+                end_block.checked_sub(&start_block) >= Some(T::MinAuctionDuration::get()),
+                Error::<T>::AuctionTimelineLowerThanMinDuration
+            );
+            // ensure buffer period is valid
+            ensure!(
+                start_block.checked_sub(&current_block) >= Some(T::MinAuctionBuffer::get()),
+                Error::<T>::AuctionTimelineBeforeBuffer
+            );
+
+            // ensure the buy_it_price is greater than start_price
+            match buy_it_price {
+                Some(price) => ensure!(price > start_price, Error::<T>::AuctionPricingInvalid),
+                None => (),
+            }
+
+            // fetch the data of given nftId
+            let nft_data = T::NFTHandler::get_nft(nft_id).ok_or(Error::<T>::NFTIdInvalid)?;
+
+            // ensure the caller is the owner of NFT
+            ensure!(nft_data.owner == creator.clone(), Error::<T>::NftNotOwned);
+
+            // ensure the nft is not listed for sale
+            ensure!(
+                nft_data.listed_for_sale == false,
+                Error::<T>::NFTAlreadyListedForSale
+            );
+
+            // ensure the nft is not in transmission
+            ensure!(
+                nft_data.in_transmission == false,
+                Error::<T>::NFTInTransmission
+            );
+
+            // Ensure origin is allowed to sell nft on given marketplace
+            T::MarketplaceHandler::is_allowed_to_list_on_marketplace(
+                marketplace_id,
+                creator.clone(),
+            )?;
+
+            // Mark NFT as listed for sale
+            T::NFTHandler::set_listed_for_sale(nft_id, true)?;
+
+            let bidders: BidderList<T::AccountId, BalanceOf<T>> = BidderList::new();
+
+            // Add auction to storage
+            Auctions::<T>::insert(
+                nft_id,
+                AuctionData {
+                    creator: creator.clone(),
+                    start_block,
+                    end_block,
+                    start_price,
+                    buy_it_price,
+                    bidders,
+                    marketplace_id,
+                    state: AuctionState::Pending,
+                },
+            );
+
+            // Emit AuctionCreated event
+            Self::deposit_event(Event::AuctionCreated {
+                nft_id,
+                marketplace_id,
+                creator,
+            });
+            // Return a successful DispatchResultWithPostInfo
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        #[transactional]
+        pub fn cancel_auction(origin: OriginFor<T>, nft_id: NFTId) -> DispatchResultWithPostInfo {
+            let who = ensure_signed(origin)?;
+            let current_block = frame_system::Pallet::<T>::block_number();
+
+            // fetch the data of given nftId
+            let nft_data = T::NFTHandler::get_nft(nft_id).ok_or(Error::<T>::NFTIdInvalid)?;
```

Comment:
It's a unnecessary check. Use the creator field from auction struct to check if the caller is authorized to cancel the auction.